test/test_intensify/test_intensify.py

```diff
@@ -449,28 +473,25 @@ def target(x):
         # intensification iteration #2
         config, _ = intensifier.get_next_challenger(challengers=None,  # don't need a new list here as old one is cont'd
                                                     chooser=None)
-        self.assertEqual(config, self.config1)
         self.assertEqual(intensifier.stage, IntensifierStage.RUN_INCUMBENT)
         inc, _ = intensifier.eval_challenger(challenger=config, incumbent=inc, run_history=self.rh, )
 
-        # incumbent is run on the 2nd instance
-        self.assertEqual(inc, self.config3)
-        self.assertEqual(self.stats.inc_changed, 3)
+        self.assertEqual(inc, self.config1)
         self.assertEqual(intensifier.stage, IntensifierStage.RUN_CHALLENGER)
 
-        self.assertEqual(len(self.rh.get_runs_for_config(self.config1, only_max_observed_budget=True)), 1)
-        self.assertEqual(len(self.rh.get_runs_for_config(self.config2, only_max_observed_budget=True)), 1)
-        self.assertEqual(len(self.rh.get_runs_for_config(self.config3, only_max_observed_budget=True)), 2)
+        self.assertEqual(len(self.rh.get_runs_for_config(self.config1, only_max_observed_budget=True)), 3)
+        self.assertEqual(len(self.rh.get_runs_for_config(self.config2, only_max_observed_budget=True)), 2)
+        self.assertEqual(len(self.rh.get_runs_for_config(self.config3, only_max_observed_budget=True)), 0)  # capped
```

Comment:
I'm wondering if a capped run should be excluded from this function. It might be surprising to not get such runs when asking for them. @mlindauer @KEggensperger do you have any suggestions?